it/uk/gov/hmrc/excisemovementcontrolsystemapi/repository/ExciseNumberQueueWorkItemRepositoryItSpec.scala

```diff
@@ -0,0 +1,156 @@
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+import uk.gov.hmrc.excisemovementcontrolsystemapi.config.AppConfig
+import uk.gov.hmrc.excisemovementcontrolsystemapi.repository.model.ExciseNumberWorkItem
+import uk.gov.hmrc.mongo.TimestampSupport
+import uk.gov.hmrc.mongo.test.{CleanMongoCollectionSupport, PlayMongoRepositorySupport}
+import uk.gov.hmrc.mongo.workitem.ProcessingStatus.{InProgress, ToDo}
+import uk.gov.hmrc.mongo.workitem.WorkItem
+
+import java.time.Instant
+import scala.concurrent.ExecutionContext
+
+class ExciseNumberQueueWorkItemRepositoryItSpec extends PlaySpec
+  with CleanMongoCollectionSupport
+  with PlayMongoRepositorySupport[WorkItem[ExciseNumberWorkItem]]
+  with IntegrationPatience
+  with BeforeAndAfterEach
+  with BeforeAndAfterAll
+  with OptionValues
+  with GuiceOneAppPerSuite {
+
+  protected implicit val ec: ExecutionContext = ExecutionContext.Implicits.global
+  private val appConfig = app.injector.instanceOf[AppConfig]
+  private val dateTimeService = mock[TimestampSupport]
+  private val timestamp = Instant.parse("2018-11-30T18:35:24.00Z")
+
+  protected override val repository = new ExciseNumberQueueWorkItemRepository(
+    appConfig,
+    mongoComponent,
+    dateTimeService
+  )
+
+  protected def appBuilder: GuiceApplicationBuilder =
+    new GuiceApplicationBuilder()
+      .configure(
+        "mongodb.uri" -> mongoUri
+      )
+
+  override implicit lazy val app: Application = appBuilder.build()
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+
+    when(dateTimeService.timestamp()).thenReturn(timestamp)
+  }
+
+  override def afterAll(): Unit = {
+    super.afterAll()
+    dropDatabase()
+  }
+
+  "getWorkItemForErn" should {
+
+    "return Work Item with that ern if one exists in the db " in {
+      val expectedWorkItem = insertWorkItemForErn("ern123")
+
+      val result = repository.getWorkItemForErn("ern123").futureValue
+
+      result mustBe Some(expectedWorkItem)
+    }
+
+    "return empty list if that ern does not exist in db" in {
+
+      val result = repository.getWorkItemForErn("ern124").futureValue
+
+      result mustBe None
+    }
+
+  }
+
+  "updateWorkItem" should {
+
+    "update Work Item in db" in {
+
+      val dbWorkItem = insertWorkItemForErn("ern1")
+
+      //Want to make sure the updatedAt is...updated. It is already set to "timestamp" so use a new one
+      val newTimestamp = Instant.parse("2023-11-23T16:00:00.00Z")
+      when(dateTimeService.timestamp()).thenReturn(newTimestamp)
+
+      val updatedWI = dbWorkItem.copy(
+        item = dbWorkItem.item.copy(fastPollRetriesLeft = 23),
+        availableAt = Instant.parse("2023-11-23T15:25:21.12Z"),
+        receivedAt = timestamp.plusSeconds(10),
+        status = InProgress,
+        failureCount = 18
+      )
+
+      repository.saveUpdatedWorkItem(updatedWI).futureValue mustBe true
+
+      val savedWorkItems = repository.getWorkItemForErn("ern1").futureValue
```

Comment:
here again I would use the Mongo support function. Using the same repository that we are testing for confirm the results, may result in brittle test as if the getWorkItemForErn function has a bug then this test wil fail for the wrong reason

Reply:
Updated